index.html

```diff
@@ -49,6 +70,55 @@
 
       var selectedConfigs = new Set();
 
+      function loadVersions(data) {
+      }
+
+      function navigateToVersion(version) {
+        if(!version) {
+          return // if no version present, do not navigate
+        }
+        let params = new URLSearchParams(document.location.search);
+        params.delete("version");
+        params.append("version", version);
+        url = window.location.pathname + `?${params.toString()}`;
+        window.location.href = url;
+      }
+
+      function populateVersionOptions(data) {
+        data = JSON.parse(data)
+        let resp = Array.from(data);
+        let versions = resp.map((res)=>res.sha)
+        const vms = document.getElementById("versionModalSelector");
+        versions.forEach(version=>{
+          const optionElement = document.createElement('option');
+          optionElement.value = optionElement.textContent = version;
+          vms.append(optionElement);
+        })
+
+      }
+
+      function getSummaryURL(version='') {
+        if(!version) return "summary.json" // By default fetch latest configuration summary
+        return `https://raw.githubusercontent.com/oracle/kconfigs/${version}/docs/summary.json`
+      }
+
+      function updateVersion(invalid=false) {
+        let params = new URLSearchParams(document.location.search);
+        let sp = document.getElementById("heading")
+        if(invalid) {
+          sp.innerHTML = `Distribution Kernel Configs`
+          alert("Invalid Version. Please add a valid version")
+          params.delete("version")
+          url = window.location.pathname + (params.toString())?`?${params.toString()}`:''
+          window.history.pushState({}, '', url)
+          return
+        }
+        var version = params.get("version")
+        if(!version) return
```

Comment:
This statement causes `version` to be `undefined` at initialization. This means that when the page is loaded, the page sends an `alert` saying "invalid version". We don't really want that. A URL without the `version` query parameter should default to the most recent version, i.e. `gh-pages`.
```suggestion
        var version = params.get("version");
        if(!version) return "gh-pages";
```